components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/internal/ElasticFunctionsServiceComponent.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.elk.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.base.api.ServerConfigurationService;
+import org.wso2.carbon.identity.application.authentication.framework.JsFunctionRegistry;
+import org.wso2.carbon.identity.conditional.auth.functions.elk.CallElasticFunction;
+import org.wso2.carbon.identity.conditional.auth.functions.elk.CallElasticFunctionImpl;
+import org.wso2.carbon.identity.conditional.auth.functions.elk.ElasticAnalyticsEngineConfigImpl;
+import org.wso2.carbon.identity.core.util.IdentityCoreInitializedEvent;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.governance.common.IdentityConnectorConfig;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+/**
+ * OSGi declarative services component which handle ELK related conditional auth functions.
+ */
+
+@Component(
+        name = "identity.conditional.auth.functions.elk.component",
+        immediate = true
+)
+public class ElasticFunctionsServiceComponent {
+
+    public static final String FUNC_CALL_ELASTIC = "callElastic";
+    public static final String SECURITY_TRUSTSTORE_LOCATION = "Security.TrustStore.Location";
+    public static final String SECURITY_TRUSTSTORE_TYPE = "Security.TrustStore.Type";
+    public static final String SECURITY_TRUSTSTORE_PASSWORD = "Security.TrustStore.Password";
+
+    private static final Log LOG = LogFactory.getLog(ElasticFunctionsServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+        try {
```

Comment:
Add a new line here.

Reply:
Resolved in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/110